### PR TITLE
tidb-lightning: rename tables and databases

### PR DESCRIPTION
**Pull request by pepezzzz**
add rename section

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

  lightning route function doesn't work, especially with local mode.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/tidb-lightning-data-source.md [hunk 1]
```diff
@@ -26,6 +26,17 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+## 表库重命名
+
+TiDB Lightning 运行时按照文件的命令规则导入到对应的表库位置。如果数据库和表名的位置发生变化，推荐重命名文件后再进行导入。
+RedHat-Like Linux 可以使用以下的 `rename` 命令对 `data-source-dir` 目录下的文件进行批量重命名。
+
+    ```bash
+    rename srcdb. tgtdb. *.sql
+    ```
+
+文件中的数据库名修改后，推荐删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 Schema 文件。如果是表名进行修改，还需要修改包含表名的 `CREATE TABLE` DDL 语句的 Schema 文件。
```
Comment (lance6716) [inline tidb-lightning/tidb-lightning-data-source.md:38]: 删除的话，需要提一下手动建表

**Review by lance6716 — APPROVED** [review]
rest lgtm

File: tidb-lightning/tidb-lightning-data-source.md (hunk 1 above)
Comment (pepezzzz) [reply tidb-lightning/tidb-lightning-data-source.md:38]: 库的 DDL 现在的版本删除后会自动建。表的 DDL 只修改不删除。

**Review by okJiang — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by Frank945946 — APPROVED** [review]

File: tidb-lightning/tidb-lightning-data-source.md [hunk 2]
```diff
@@ -304,14 +372,14 @@ TiDB Lightning 仅识别符合命名要求的数据文件，但在某些情况
 
 通常 `data-source-dir` 会被配置为`S3://some-bucket/some-subdir/some-database/` 以导入 `some-database` 库。
 
-根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
+根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]/)([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]/)(?:[a-z0-9\-_.]+.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:375]:
```suggestion
根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 3]
```diff
@@ -272,7 +340,7 @@ TiDB Lightning 目前仅支持由 Amazon Aurora 或者 Hive 导出快照生成
 ```
 [[mydumper.files]]
 # 解析 AWS Aurora parquet 文件所需的表达式
-pattern = '(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
+pattern = '(?i)^(?:[^/]/)([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]/)(?:[a-z0-9\-_.]+.(parquet))$'
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:343]:
```suggestion
pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$'
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 4]
```diff
@@ -304,14 +372,14 @@ TiDB Lightning 仅识别符合命名要求的数据文件，但在某些情况
...
-根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
+根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]/)([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]/)(?:[a-z0-9\-_.]+.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
 
 根据上述正则表达式及相应的 index 编写配置文件，TiDB Lightning 即可识别非默认命名规则的文件，最终实际配置如下：
 
 ```
 [[mydumper.files]]
 # 解析 AWS Aurora parquet 文件所需的表达式
-pattern = '(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
+pattern = '(?i)^(?:[^/]/)([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]/)(?:[a-z0-9\-_.]+.(parquet))$'
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:382]:
```suggestion
pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$'
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 5]
```diff
@@ -272,7 +340,7 @@ TiDB Lightning 目前仅支持由 Amazon Aurora 或者 Hive 导出快照生成
 ```
 [[mydumper.files]]
 # 解析 AWS Aurora parquet 文件所需的表达式
-pattern = '(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
+pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$'
```
Comment (okJiang) [inline tidb-lightning/tidb-lightning-data-source.md:343]:
```suggestion
pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
```

why changed `+\`?

File: tidb-lightning/tidb-lightning-data-source.md [hunk 6]
```diff
@@ -304,14 +372,14 @@ TiDB Lightning 仅识别符合命名要求的数据文件，但在某些情况
 
 通常 `data-source-dir` 会被配置为`S3://some-bucket/some-subdir/some-database/` 以导入 `some-database` 库。
 
-根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
+根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
```
Comment (okJiang) [inline tidb-lightning/tidb-lightning-data-source.md:375]: ditto

File: tidb-lightning/tidb-lightning-data-source.md [hunk 7]
```diff
@@ -304,14 +372,14 @@ TiDB Lightning 仅识别符合命名要求的数据文件，但在某些情况
...
 
 根据上述正则表达式及相应的 index 编写配置文件，TiDB Lightning 即可识别非默认命名规则的文件，最终实际配置如下：
 
 ```
 [[mydumper.files]]
 # 解析 AWS Aurora parquet 文件所需的表达式
-pattern = '(?i)^(?:[^/]*/)*([a-z0-9_]+)\.([a-z0-9_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
+pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]\+.(parquet))$'
```
Comment (okJiang) [inline tidb-lightning/tidb-lightning-data-source.md:382]: ditto

File: tidb-lightning/tidb-lightning-data-source.md (hunk 5 above)
Comment (okJiang) [reply tidb-lightning/tidb-lightning-data-source.md:343]: @hfxsd

File: tidb-lightning/tidb-lightning-data-source.md (hunk 6 above)
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:375]:
```suggestion
根据上述 Parquet 文件的路径，你可以编写正则表达式 `(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$`，得到的 match group 中 index=1 的内容为 `some-database`，index=2 的内容为 `some-table`，index=3 的内容为 `parquet`。
```

File: tidb-lightning/tidb-lightning-data-source.md (hunk 7 above)
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:382]:
```suggestion
pattern = '(?i)^(?:[^/]*/)*([a-z0-9\-_]+).([a-z0-9\-_]+)/(?:[^/]*/)*(?:[a-z0-9\-_.]+\.(parquet))$'
```

**Review by hfxsd — APPROVED** [review]

File: tidb-lightning/tidb-lightning-data-source.md [hunk 8]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+
+### 批量重命名文件
+
+RedHat-Like Linux 可以使用以下的 `rename` 命令对 `data-source-dir` 目录下的文件进行批量重命名。
+
+```shell
+rename srcdb. tgtdb. *.sql
+```
+
+文件中的数据库名修改后，建议删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 `schema-create.sql` 文件。如果修改的是表名，还需要修改包含表名的 `CREATE TABLE` DDL 语句的 `schema.sql` 文件。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:41]:
```suggestion
修改了文件中的数据库名后，建议删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 `schema-create.sql` 文件。如果修改的是表名，还需要修改包含表名的 `CREATE TABLE` DDL 语句的 `schema.sql` 文件。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 9]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+
+### 使用正则表达式在线替换名称
+
+要使用正则表达式在线替换名称，你需要在 [[mydumper.files]] 内使用 `pattern` 匹配文件名，将 `schema` 和 `table` 换成目标名。请参考[自定义文件匹配](#自定义文件匹配)。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:45]:
```suggestion
要使用正则表达式在线替换名称，你需要在 `[[mydumper.files]]` 配置中使用 `pattern` 匹配文件名，将 `schema` 和 `table` 换成目标名。具体配置请参考[自定义文件匹配](#自定义文件匹配)。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 10]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+
+下面是使用正则表达式在线替换名称的示例。其中：
+
+- 数据文件 `pattern` 的匹配规则是 '^({schema_regrex})\.({table_regrex})\.({file_serial_regrex})\.(csv|parquet|sql)'。
+- `schema` 可以指定为 '$1'，代表第一个正则表达式 `schema_regrex` 取值不变，或者是一个字符串，如 'tgtdb'，代表固定的目标数据库。
+- `table` 可以指定为 '$2'，代表第二个正则表达式 `table_regrex` 取值不变，或者是一个字符串，如 't1'，代表固定的目标表。
+- `type` 可以指定为 '$3'，代表是数据文件类型，`"table-schema"` 代表是 `schema.sql` 文件，或 `"schema-schema"` 代表是 `schema-create.sql` 文件。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:52]:
```suggestion
- `schema` 可以指定为 `'$1'`，代表第一个正则表达式 `schema_regrex` 取值不变；`schema` 也可以指定为一个字符串，如 `'tgtdb'`，代表固定的目标数据库名。
- `table` 可以指定为 `'$2'`，代表第二个正则表达式 `table_regrex` 取值不变；`table` 也可以指定为一个字符串，如 `'t1'`，代表固定的目标表名。
- `type` 可以指定为 `'$3'`，代表数据文件类型；`type` 可以指定为 `"table-schema"` （代表 `schema.sql` 文件） 或 `"schema-schema"`（`schema-create.sql` 文件）。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 11]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+### 使用正则表达式在线替换名称
+
+要使用正则表达式在线替换名称，你需要在 `[[mydumper.files]]` 配置中使用 `pattern` 匹配文件名，将 `schema` 和 `table` 换成目标名。具体配置请参考[自定义文件匹配](#自定义文件匹配)。
+
+下面是使用正则表达式在线替换名称的示例。其中：
+
+- 数据文件 `pattern` 的匹配规则是 '^({schema_regrex})\.({table_regrex})\.({file_serial_regrex})\.(csv|parquet|sql)'。
+- `schema` 可以指定为 `'$1'`，代表第一个正则表达式 `schema_regrex` 取值不变；`schema` 也可以指定为一个字符串，如 `'tgtdb'`，代表固定的目标数据库名。
+- `table` 可以指定为 `'$2'`，代表第二个正则表达式 `table_regrex` 取值不变；`table` 也可以指定为一个字符串，如 `'t1'`，代表固定的目标表名。
+- `type` 可以指定为 `'$3'`，代表数据文件类型；`type` 可以指定为 `"table-schema"` （代表 `schema.sql` 文件） 或 `"schema-schema"`（`schema-create.sql` 文件）。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:52]:
```suggestion
- `type` 可以指定为 `'$3'`，代表数据文件类型；`type` 可以指定为 `"table-schema"`（代表 `schema.sql` 文件） 或 `"schema-schema"`（代表 `schema-create.sql` 文件）。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tidb-lightning/tidb-lightning-data-source.md [hunk 12]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+table = '$2'
+type = "table-schema"
+[[mydumper.files]]
+pattern = '^(srcdb)\.(.*?)\.(?:[0-9]+)\.(csv|parquet|sql)'
+schema = 'tgtdb'
+table = '$2'
+type = '$3'
+```
+
+如果是使用 `gzip` 方式备份的数据文件，需要对应地配置压缩格式。数据文件 `pattern` 的匹配规则是 '^({schema_regrex})\.({table_regrex})\.({file_serial_regrex})\.(csv|parquet|sql)\.(gz)'。`compression` 可以指定为 '$4' 代表是压缩文件格式。示例如下：
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:73]:
```suggestion
如果是使用 `gzip` 方式备份的数据文件，需要对应地配置压缩格式。数据文件 `pattern` 的匹配规则是 `'^({schema_regrex})\.({table_regrex})\.({file_serial_regrex})\.(csv|parquet|sql)\.(gz)'`。`compression` 可以指定为 `'$4'` 代表是压缩文件格式。示例如下：
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 13]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+
+### 批量重命名文件
+
+如果你使用的是 Red Hat Linux 或基于 Red Hat 的 Linux 发行版，可以使用以下的 `rename` 命令对 `data-source-dir` 目录下的文件进行批量重命名：
+
+```shell
+rename srcdb. tgtdb. *.sql
+```
+
+修改了文件中的数据库名后，建议删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 `schema-create.sql` 文件。如果修改的是表名，还需要修改包含表名的 `CREATE TABLE` DDL 语句的 `schema.sql` 文件。
```
[the remark this answers is not in the document]
Comment (pepezzzz) [reply tidb-lightning/tidb-lightning-data-source.md:41]: 是的。

File: tidb-lightning/tidb-lightning-data-source.md [hunk 14]
```diff
@@ -26,6 +26,74 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
...
+
+### 批量重命名文件
+
+如果你使用的是 Red Hat Linux 或基于 Red Hat 的 Linux 发行版，可以使用 `rename` 命令对 `data-source-dir` 目录下的文件进行批量重命名。例如：
+
+```shell
+rename srcdb. tgtdb. *.sql
+```
+
+修改了文件中的数据库名后，建议删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 `schema-create.sql` 文件。如果修改的是表名，还需要修改包含表名的 `CREATE TABLE` DDL 语句的 `schema.sql` 文件。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:41]:
```suggestion
修改了文件中的数据库名后，建议删除 `data-source-dir` 目录下包含 `CREATE DATABASE` DDL 语句的 `${db_name}-schema-create.sql` 文件。如果修改的是表名，还需要修改包含 `CREATE TABLE` DDL 语句的 ${db_name}.${table_name}-schema.sql` 文件中的表名。
```

**Pull request merged**